### DISCUSSION TITLE
Is there a multi-output or multi-label example for graph classification?

### DISCUSSION THREAD

**Discussion by Boehm92** [pyg-team/pytorch_geometric#4268] [Q&A]
Hi,
i'm working on a GNN to classify 3D geometric shapes in a 3D cube. The cube and its geometric shape is mesh based. 

I'm allready capable to classify a cube with one geometric shape and it works realy well. For that i'm using the dataset from this paper https://github.com/zibozzb/FeatureNet and this PyG examle https://colab.research.google.com/drive/1I8a0DfQ3fI7Njc62__mVXUlcAleUclnb?usp=sharing

My graphs look something like that: `Data(x=[14, 3], edge_index=[2, 144], y=[24])`

Now, i want to classify a cube with two or more shapes. I tried following steps:
- reconstructed the target to an array with 24 element: `y = [0, 0, 1, 0, ..., 0, 1]`
- switched BCE to BCEWithLogitsLoss.

But i'm not shure how to continue with the training method or if this is the right way anyway. 

Is there some example for multi-output or multi-label classification?

Kind Regards 
Stefan

**Comment by rusty1s** [comment]
The [`PPI` example](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/ppi.py) denotes a multi-label classification problem (although it is not mesh-based). It uses `torch.nn.BCEWithLogitsLoss()` for loss computation.
